source/code/go/src/plugins/utils.go

```diff
@@ -68,7 +69,20 @@ func CreateHTTPClient() {
 	}
 
 	tlsConfig.BuildNameToCertificate()
-	transport := &http.Transport{TLSClientConfig: tlsConfig}
+	transport := &http.Transport{TLSClientConfig: tlsConfig}	
+	// set the proxy if the proxy configured
+	if ProxyEndpoint != "" {
+		proxyEndpointUrl, err := url.Parse(ProxyEndpoint)		
+		if err != nil {
+			message := fmt.Sprintf("Error parsing Proxy endpoint %s", err.Error())
+			SendException(message)
+			time.Sleep(30 * time.Second)
```

Comment:
No sleep. Proxy uri is a free form input coming from the customer. If it is incorrect (or in a format that we dont understand), how will the customer know that the issue is with proxy due to which the container keeps restarting and they need to fix the proxy uri ?

Reply:
> No sleep. Proxy uri is a free form input coming from the customer. If it is incorrect (or in a format that we dont understand), how will the customer know that the issue is with proxy due to which the container keeps restarting and they need to fix the proxy uri ?
> proxy will be validated in the base agent and also will add the helm client to do the validation. customer should be seeing onboarding failures (helm chart installation and logs of agent pods ) if the proxy endpoint is invalid similar how they see if the workspace is invalid.